grout/lookups.py

```diff
@@ -100,12 +100,23 @@ def sql(self):
 
             # The check on 'pattern' here allows us to apply a pattern filter on top of others
             if 'pattern' in rule:
-                # Don't filter as an exact match on the text entered; match per word.
-                for pattern in shlex.split(rule['pattern']):
-                    if rule['_rule_type'] == 'containment_multiple':
-                        sql_tuple = FilterTree.text_similarity_filter(path, pattern, True)
-                    else:
-                        sql_tuple = FilterTree.text_similarity_filter(path, pattern, False)
+                # Filter word by word individually, but make sure to keep quoted words together
+                # Regex to match quoted substrings from https://stackoverflow.com/a/5696141:
+                # /"[^"\\]*(?:\\.[^"\\]*)*"/g
+                word_regex =  r'({})'.format(
+                    '|'.join([
+                        # Match groups of words in double quotes (Allowing for escaping)
+                        '\\"[^\\"\\\\]*(?:\\\\.[^\\"\\\\]*)*\\"',
```

Comment:
If we prefix these strings with `r`, then we won't have to worry about escaping backslashes and the regex will be more readable:
```python
>>> r'"[^"\\]*(?:\\.[^"\\]*)*"'
'"[^"\\\\]*(?:\\\\.[^"\\\\]*)*"'
```

Reply:
Good call, done!